packages/react-native/src/compile.ts

```diff
@@ -0,0 +1,244 @@
+import execa from 'execa';
```

Comment:
the compile.ts module is a duplicate of the typescript compiler. Instead of duplicating the compile module we need to make it more abstract to get all information from the preset and not from the module. This will help reusing the code in the future.

Reply:
Please consult if you need to change the preset interface.

Reply:
You right, thanks for your feedback.
I'll do these changes and PR again.